### DISCUSSION TITLE
Can cells be deleted due to divison by zero?

### DISCUSSION THREAD

**Discussion by ErenC1** [UvaCsl/HemoCell#93] [General discussion]
Hi!

I am facing a lot of deleted cells in these days (my channels are reaaaally narrow), I went back to the original paper and wrote a simple MATLAB script to investigate it in more detail. I am pretty sure they are not deleted because of force cap, because I increased it, and my flow rate is really low.
 I adjusted my parameters according to a paper(kBend, kLink etc.), but I am still getting deleted cells, I am pretty sure tau values are the reason behind this. Let's say I used pi/2 (or pi/6, both is not working for me) for tauBend, in this case what's gonna happen when cell starts to bend and dtheta =pi/2 (or pi/6)? Does cell gets deleted due to the limiting tau? or similar values for tauLink, tauArea etc. 
 
 edit:
 Like in this example, after the 3rd state this RBC deleted. 
[https://imgur.com/a/9Kyhdyz](url)
 
 Thank you in advance,
 Eren.

**Comment by mojtabaAmiraslanpour** [comment]
Hey Eren,

In my case, this issue would occur when RBCs hit the wall. Because in LBM there is no actual solid wall there but we are using the bounce-back BC. In this case, RBCs can penetrate the wall and eventually be deleted. In your case, I see you have a very narrow opening there which may be the problem. Maybe you can try it with very slow velocity to give it sometime to deform and note penetrate the wall and see if that works.

**Reply by ErenC1** [reply]
Hi Mojtaba,

Actually, since HemoCell is optimized for high shear flows, this RBC can pass through the slit in high velocities but cannot pass through the slit in low velocities (it sounds insane, but that's the case). I am pretty sure its about tau's, and I need another set of parameters for this case. I just need to hear a validation from Dr. Závodszky...

**Comment by gzavo** [comment]
Hi,

Some first ideas based on the image:
This scenario is indeed extreme squeezing that pushes the capabilities of HemoCell, primarily due to the fluid resolution (0.5 um).
I would check are streamlines (e.g., ParaView -> Stream Tracer filter). I suspect at high flow they align with the slit already at some distance so the RBC arrives "pre-deformed" at the slit, while in low flow case the whole deformation happens right in the slit. This in turn might generate momentarily higher forces.
Sticking in or to the wall should not happen. The current implementation of IBM in HemoCell disregards wall cells when interpolating the neighboring velocities. However, if the slit is only ~2 lattice unit wide, that might cause a numerical challenge. Since all the surface particles of the RBC will take their velocity from those same few fluid lattices, they will not be able to move independently.

There are other groups using HemoCell with similar narrow geometries:
[https://doi.org/10.1007/s10404-020-02337-3](https://doi.org/10.1007/s10404-020-02337-3) 
Some are modifying the cell membrane:
[https://doi.org/10.1007/s10237-022-01567-4](https://doi.org/10.1007/s10237-022-01567-4)

**Reply by ErenC1** [reply]
Hi,

Sorry about not giving enough information. This slit is 3 LU wide and the velocity should not be the problem, because in each case (this one, and my narrow capillaries where flow area is usually around 5 x 5, 4 x 7 LU etc. ) when I increase the driving force cells stops getting deleted.

Such as, in the case I showed above Ubar = 50 um/s (average velocity before the slit) when I increase it to 0.02 m/s, RBC passes through the slit, see the below image. I would expect it to be deleted at the extreme velocity, but it passes through (with a great deformation), whereas it gets deleted at low velocities.

Thank you for the papers, do you advise that increasing resolution might help here? Because in both papers are RBC resolution is much higher, if I increase L0 2x, I need to increase dx 2x too, that means my dt will probably will go as low as 1e-10. I don't know how they used dt=1e-7 in the first paper, my dt is usually around 1e-8 for tau=1 and, capillary pressure I got from physiology books. Decreasing dx will result in much lower dt...

[https://imgur.com/a/5moHY6Z](url)

Additional image: when RBC placed with 0 angle, it passes through and results are close to the reference paper (for low velocity).
https://imgur.com/a/0ifQXM5

**Reply by ErenC1** [reply]
Update on resolution, I lowered dx to 0.25 um, after 48 hrs computation with 32 cores single a RBC moved only 2 um(in above configuration), as its expected dt is around 1e-10. I don't think this is feasible

**Reply by ErenC1** [reply]
@gzavo Hello again! You were right, at high shear deformation happens before the slit, that's why it can pass like that. I eventually handled the situation, RBCs can freely enter the slit and computational times are acceptable now (Thnx to Dr. Krüger) but there is one problem left, that is RBC shows really "sharp" behavior when entering the slit. Following links shows the RBCs when they enter the slit.
https://drive.google.com/file/d/1uBhNGm_vUktZnc6zJn-ccIo9CK9cLrlM/view?usp=share_link
https://drive.google.com/file/d/1sjHf8rQkCNIbteCEB1-hDa5eaR8z_lzQ/view?usp=share_link
I can play with kB and ks since they are correlated and ks is related with capillary number but IDK how to deal with kA and kV. Do you have any suggestions for me to check out?

Here is the paper I used in validation, you can see that their RBC is not really sharp before the slit:
https://link.springer.com/article/10.1007/s10237-016-0861-7

**Reply by mojtabaAmiraslanpour** [reply]
Hi Eren, can you please explain how you managed to handle the problem?
Thanks.

**Reply by ErenC1** [reply]
Hi Mojtaba, and @LGM-1 ,

Unfortunately I couldn't fully solve this problem yet. And it looks like my PhD is depending on this...
As you can see from this image, some of the particles sticks to the wall even with the "high" repulsion force. High shear near the wall is the problem here.
https://drive.google.com/file/d/1mPEW9ODDNUGMk86GFP81KtYMDRzb-Ln2/view?usp=sharing

I tested it with "high force" cell stretch and you can see the result below. RBC is going craaaazy, and eventually it is deleted.
https://drive.google.com/file/d/1SPm8MelWjIVyn4V4QWewJ04_QCRI8qmQ/view?usp=sharing

So, keeping RBCs away from the wall is the best choice, but it is kinda impossible for me. I'm looking at the alternative dynamics like SpecularReflection but it didn't work too. I'll try a "small" wall velocity that doesn't really affect the physics but it can make it possible to simulate RBCs in small channels. If it doesn't work IDK what I can do for now. Maybe @gzavo can give us some idea?

Best,
Eren.

**Reply by gzavo** [reply]
Hi @ErenC1,

In the video on the cell stretching I think the issue is the flow field. My guess is that it has exploded numerically, that is why you get weird shapes, as the cell can't move properly anymore through interaction with fluid. If you plot the fluid field you will probably see a bunch if extreme large or NaN values. It means the forcing is too large, and you have accelerated the fluid too much in one iteration. This has to do with the properties of LBM. The only thing that you can easily do is to reduce dt. Perhaps also in the small channel the appearing forces are too high for the LBM to handle at the given time resolution. I would look at the flow field first.
Repulsive particles in the wall might help in cases with e.g., sharp edges, but not if you use them throughout the small capillary, since then they just reduce the effective diameter further.

Also, the fluid-structure coupling, IBM, in general is not very good in resolving close contacts, so when you previously wrote 3 LU (~1,5 um) slit, I don't think that is possible with the current dx=0.5 um. Smaller dx might work, but then you also need to increase RBC resolution and it will increase computational costs significantly.

**Comment by LGM-1** [comment]
Hey Eren：
Sorry to bother you, I want to ask how you solved this last question I seem to have the same problem as you, when the cell is very slow through the gap it will be deleted and disappear
